ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -158,6 +147,10 @@ private void initReplDump(ASTNode ast) throws HiveException {
         if (null != replConfigs) {
           for (Map.Entry<String, String> config : replConfigs.entrySet()) {
             conf.set(config.getKey(), config.getValue());
+            if ("hive.repl.dump.metadata.only".equalsIgnoreCase(config.getKey()) &&
```

Comment:
Shall use HiveConf.getBoolVar outside the for loop to check this config.

Reply:
done